master/buildbot/changes/gitpoller.py

```diff
@@ -159,8 +170,16 @@ def poll(self):
             '+%s:%s' % (self._removeHeads(branch), self._trackerBranch(branch))
             for branch in branches
         ]
-        yield self._dovccmd('fetch',
-                            [self.repourl] + refspecs, path=self.workdir)
+
+        while True:
+            try:
+                yield self._dovccmd('fetch', [self.repourl] + refspecs,
+                                    path=self.workdir)
+            except GitError, e:
+                log.msg(e)
+                time.sleep(self.pollInterval)
```

Comment:
ah ah you felt in my trap!
http://stackoverflow.com/questions/9310350/twisted-non-blocking-method-how-to

```
        yield util.asyncSleep(self.pollInterval)

```